base/sim/fastsim/FairFastSimRunConfiguration.cxx

```diff
@@ -40,6 +41,9 @@ FairFastSimRunConfiguration::~FairFastSimRunConfiguration()
 
 TG4VUserFastSimulation* FairFastSimRunConfiguration::CreateUserFastSimulation()
 {
-    LOG(info) << "Going to create FairFastSimulation";
-    return new FairFastSimulation();
+    if (fUseFastSim) {
+        LOG(info) << "Going to create FairFastSimulation";
+        return new FairFastSimulation();
+    }
+    return 0;
```

Comment:
Do you have an example of how `CreateUserFastSimulation()` is used?
What happens when `0` is returned (it should really be `nullptr`)?
Does the caller have to check for `nullptr` and if it occurs create `TG4RunConfiguration` instead?

The proposal from the related issue looks more reasonable to me (albeit more verbose on the caller side).
It doesn't seem to make sense to create `FairFastSimRunConfiguration` in the first place, if `FairFastSim_UseFastSim` is false.

Reply:
The test shows that there is no difference between using TG4RunConfiguration
and FairFastRunSimConfiguration without FairFastSimulation.
I agree with you though, that it does make more sense to use the original
TG4RunConfiguration in case of standard simulation.

Reply:
To be honest, I do not think we should add the proposed `FairFastSim_UseFastSim` option. Also the proposed `void FairFastSimRunConfiguration::UseFastSim(bool flag = true)` C++ API makes no sense to me as a user.